src/fr/bigsolo/src/eu/kanade/tachiyomi/extension/fr/bigsolo/BigSolo.kt

```diff
@@ -1,10 +1,162 @@
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import okhttp3.Request
+import okhttp3.Response
+import java.net.URI
+
+class BigSolo : HttpSource() {
+
+    override val name = "BigSolo"
+    override val baseUrl = "https://bigsolo.org"
+    override val lang = "fr"
+    override val supportsLatest = true
+    override val id = 4410528266393104437
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/data/series", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val series = response.parseAs<SeriesResponse>()
+        val mangaList = mutableListOf<SManga>()
+
+        for (serie in series.reco) {
+            mangaList.add(serie.toDetailedSManga())
+        }
+
+        return MangasPage(mangaList, false)
+    }
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/data/series", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        return searchMangaParse(response)
+    }
+
+    // Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = if (query.isNotBlank()) {
+            "$baseUrl/data/series#$query"
+        } else {
+            "$baseUrl/data/series"
+        }
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val series = response.parseAs<SeriesResponse>()
+        val allSeries = (series.series + series.os).sortedByDescending { it.lastChapter?.timestamp }
+        val mangaList = mutableListOf<SManga>()
+
+        val fragment = response.request.url.fragment
+        val searchQuery = fragment ?: ""
+
+        for (serie in allSeries) {
+            if (searchQuery.startsWith("SLUG:") && serie.slug == searchQuery.removePrefix("SLUG:")) {
+                mangaList.add(serie.toDetailedSManga())
+                break
+            }
+
+            if ((
+                searchQuery.isBlank() ||
+                    serie.title.contains(searchQuery, ignoreCase = true) ||
+                    serie.alternativeTitles.any { it.contains(searchQuery, ignoreCase = true) } ||
+                    serie.jaTitle.contains(searchQuery, ignoreCase = true)
+                ) && !searchQuery.startsWith("SLUG:")
+            ) {
+                mangaList.add(serie.toDetailedSManga())
+            }
+        }
+
+        return MangasPage(mangaList, false)
+    }
+
+    // Details
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        val splitedPath = URI(document.baseUri()).path.split("/")
+        val slug = splitedPath[1]
+        val serieJson =
+            client.newCall(GET("$baseUrl/data/series/$slug", headers))
+                .execute()
```

Comment:
You're not really using the response, so I'm tempted to suggest this should be done in `mangaDetailsRequest()` instead. This would allow you to skip a request.

Keep the same URL format in the SManga model to prevent duplicate library entries.